aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+        // cannot update whitelisted addresses if the whitelist minting period has already passed
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        assert!(whitelist_mint_config.whitelist_minting_end_time > timestamp::now_seconds(), error::permission_denied(EINVALID_UPDATE_AFTER_MINTING));
+
+        let i = 0;
+        while (i < vector::length(&wl_addresses)) {
+            let addr = *vector::borrow(&wl_addresses, i);
+            // assert that the specified address exists
+            assert!(account::exists_at(addr), error::invalid_argument(EACCOUNT_DOES_NOT_EXIST));
+            bucket_table::add(&mut whitelist_mint_config.whitelisted_address, addr, mint_limit);
+            i = i + 1;
+        };
+    }
+
+    /// Add destination tokens.
+    public entry fun add_tokens(
+        admin: &signer,
+        token_uris: vector<String>,
+        property_keys: vector<vector<String>>,
+        property_values: vector<vector<vector<u8>>>,
+        property_types: vector<vector<String>>
+    ) acquires NFTMintConfig, CollectionConfig, WhitelistMintConfig {
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
+
+        // cannot add more token uris if minting has already started
+        assert!(exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        assert!(whitelist_mint_config.whitelist_minting_start_time > timestamp::now_seconds(), error::permission_denied(EINVALID_UPDATE_AFTER_MINTING));
+
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        assert!(vector::length(&token_uris) == vector::length(&property_keys) && vector::length(&property_keys) == vector::length(&property_values) && vector::length(&property_values) == vector::length(&property_types), error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+
+        assert!(vector::length(&token_uris) + vector::length(&token_config.token_uri) <= token_config.collection_maximum || token_config.collection_maximum == 0, error::invalid_argument(EEXCEEDS_COLLECTION_MAXIMUM));
+        vector::append(&mut token_config.token_uri, token_uris);
+        vector::append(&mut token_config.property_keys, property_keys);
+        vector::append(&mut token_config.property_values, property_values);
+        vector::append(&mut token_config.property_types, property_types);
+    }
+
+    /// Mint source certificate.
+    public entry fun mint_source_certificate(
+        nft_claimer: &signer,
+        amount: u64
```

Comment:
shouldn't there be a max amount?

Reply:
If there's a max amount, we need to keep a table of <address, amount> - seems expensive

Reply:
Not sure what you guys mean by max amount. This contract has a max amount for whitelisted addresses in         `whitelisted_address: BucketTable<address, u64>,` whitelisted users can only mint up to an amount in this bucket table during the whitelisted minting phase. When it comes to the public minting phase, there is no amount limitation anymore.

Reply:
> When it comes to the public minting phase, there is no amount limitation anymore.

This ... doesn't make a lot of sense to me.